fava/api/__init__.py

```diff
@@ -157,6 +157,15 @@ def _apply_filters(self):
             self.date_first = self.filters['time'].begin_date
             self.date_last = self.filters['time'].end_date
 
+        self.upcoming_events = []
```

Comment:
This whole block should not be run on every filter, it belongs in `load_file`. It would also be a bit neater if this was a function (like `_sidebar_links`)